WireDomain/Sources/WireDomain/Notifications/Components/NotificationServiceExtensionFlow.swift

```diff
@@ -59,4 +67,153 @@ final class NotificationServiceExtensionFlow: BootstrapComponent {
     var processNotificationRequestStep: ProcessNotificationRequestStep {
         ProcessNotificationRequestStep(parent: self)
     }
+
+    public var backendEnvironment: BackendEnvironment2 {
+        BackendEnvironment2(legacyBackendEnvironment)
+    }
+
+    public var minTLSVersion: WireNetwork.TLSVersion {
+        if let version = appMainBundle.infoForKey("MinTLSVersion") {
+            TLSVersion(version) ?? .v1_2
+        } else {
+            .v1_2
+        }
+    }
+
+    public var preferredAPIVersion: WireNetwork.APIVersion? {
+        BackendInfo.preferredAPIVersion.flatMap {
+            APIVersion(rawValue: UInt($0.rawValue))
+        }
+    }
+
+    private var legacyBackendEnvironment: WireDataModel.BackendEnvironment {
+        let environmentType: EnvironmentType
+        if let override = sharedUserDefaults.string(
+            forKey: "BackendEnvironmentTypeOverrideKey"
+        ) {
+            environmentType = EnvironmentType(stringValue: override)
+        } else {
+            environmentType = EnvironmentType(userDefaults: sharedUserDefaults)
+        }
+
+        guard let backendEnvironment = WireTransport.BackendEnvironment(
+            userDefaults: sharedUserDefaults,
+            configurationBundle: backendBundle,
+            environmentType: environmentType
+        ) else {
+            fatal("Malformed backend configuration data")
+        }
+
+        return backendEnvironment
+    }
+
+    var appMainBundle: Bundle {
+        let mainBundle: Bundle
+        let runningInExtension = Bundle.main.bundlePath.hasSuffix(".appex")
+
+        if runningInExtension {
+            let extensionBundleURL = Bundle.main.bundleURL
+            let mainAppBundleURL = extensionBundleURL.deletingLastPathComponent().deletingLastPathComponent()
+            guard let bundle = Bundle(url: mainAppBundleURL) else { fatal("Failed to find main app bundle") }
+            mainBundle = bundle
+        } else {
+            mainBundle = .main
+        }
+        return mainBundle
+    }
+
+    var backendBundle: Bundle {
+        guard let backendBundlePath = appMainBundle.path(
+            forResource: "Backend",
+            ofType: "bundle"
+        ) else {
+            fatal("Could not find backend.bundle")
+        }
+
+        guard let bundle = Bundle(path: backendBundlePath) else {
+            fatal("Could not load backend.bundle")
+        }
+
+        return bundle
+    }
+}
+
+public extension BackendEnvironment2 {
+
+    init(_ legacyEnvironment: WireTransport.BackendEnvironment) {
```

Comment:
All these mappings are temporary until we implement multibackend support and remove the legacy environments).